docs/proposals/Periodic-Retrieval-of-Certificates-and-Keys-from-AKV.md

```diff
@@ -0,0 +1,121 @@
+
+### Key disabling
+
+A malicious user compromised the private key used for signing images. Bob promptly rotated the key, obtaining a new version, and disabled the compromised version. Bob configures Ratify to periodically retrieve all available versions of the key from AKV excluding any disabled versions, as a result, images signed with the disabled version fail signature verification. The build pipeline was triggered to build and sign all the images with the new version of key. As Ratify periodically retrieve all available versions of the key from AKV, it ensures successful verification of images signed with the latest key version without requiring manual configuration adjustments. By following these actions, Bob maintains security while seamlessly transitioning to the new version of key.
+
+### Immediate retrieval of certificates and keys 
+
+In urgent situations, such as when the private key is compromised, both Alice and Bob prefer not to wait for the next certificate and key retrieval cycle. They expect that images signed with compromised keys will fail validation immediately to avoid potential security attacks, and images signed with the rotated version can be validated successfully to prevent potential service downtime. To achieve this, they want to trigger Ratify to retrieve certificates and keys promptly, ensuring that the necessary updates are applied. This way, images can be verified with the correct configurations.
+
+## User experiences
+
+Typically, users do not need to manually update the configuration unless they specifically want to trigger an immediate retrieval. Ratify periodically retrieves the certificates and keys from Azure Key Vault (AKV), ensuring that it uses the most up-to-date available versions. Additionally, Ratify filters out any disabled versions of certificates and keys during this process.
+
+### Periodic retrieval of certificates and keys from AKV
+
+To periodically retrieve certificates and keys from AKV, users configure the `KeyManagementProvider` without specifying the version for certificates and keys. Users should have the ability to customize the time interval at which the retrieval process occurs, allowing them to override the default interval (e.g., 24 hours). As part of this enhancement, a new parameter named `retrievalInterval` will be introduced for AKV as the key management provider.
```

Comment:
Since we are retrieving all available versions, what the max volume we can support without being throttled by AKV or any other KMPs within a given time period?

Reply:
Resolved this comment as discussed during community meeting on Jun 12 2024. The new proposal will not require retrieving all available versions. By default, it will be the latest version and one previous version. If users want to fetch more versions, they should be aware of the throttling guides from different KMPs, which will be provided in the documentation.